### PR TITLE
Add keyboard shortcuts and enhance recording controls in FreeScribe

### PR DESCRIPTION
**Pull request by n41tik**
## Summary by Sourcery

Add keyboard shortcuts for key actions in FreeScribe and enhance recording controls with checks to prevent redundant operations. Update documentation to reflect these new shortcuts.

New Features:
- Introduce keyboard shortcuts for opening the FreeScribe extension, starting/stopping recording, and opening configuration settings.

Enhancements:
- Enhance recording controls by adding checks to prevent starting a new recording if one is already in progress and to prevent stopping a recording if none is in progress.

Documentation:
- Update the welcome page to include information about the new keyboard shortcuts.

### CODE REVIEW

**Comment by n41tik** [comment]
@sourcery-ai review

File: src/background.js [hunk 1]
```diff
@@ -29,3 +29,31 @@ chrome.contextMenus.onClicked.addListener((info, tab) => {
     openSidePanel(tab.windowId);
   }
 });
+
+chrome.commands.onCommand.addListener((command) => {
+  switch (command) {
+    case "freescribe_record_start":
+      chrome.tabs.query({ active: true, currentWindow: true }, (tabs) => {
+        openSidePanel(tabs[0].windowId);
+        setTimeout(() => {
```
[the remark this answers is not in the document]
Comment (n41tik) [reply src/background.js]: removed set time out and directly sending data

File: src/background.js [hunk 2]
```diff
@@ -29,3 +29,31 @@ chrome.contextMenus.onClicked.addListener((info, tab) => {
     openSidePanel(tab.windowId);
   }
 });
+
+chrome.commands.onCommand.addListener((command) => {
```
[the remark this answers is not in the document]
Comment (n41tik) [reply src/background.js:33]: updated the whole logic to directly send data to the script and handling it on the frontend. command will only run if the extension is open

File: src/index.js [hunk 3]
```diff
@@ -118,6 +118,10 @@ async function startMicStream() {
 }
 
 async function startRecording() {
+  if (isRecording) {
+    Logger.info("Recording already in progress");
```
[the remark this answers is not in the document]
Comment (n41tik) [reply src/index.js:122]: logging error and not throwing it

**Review by yingbull — APPROVED** [review]

**Pull request merged**